build-phar.sh

```diff
@@ -1,11 +1,19 @@
 #!/usr/bin/env bash
 
+set -euf -o pipefail
+
 ./download-box.sh
 
+function restorePlatform {
+    composer config --unset platform
+    mv -f composer.lock.back composer.lock || true
+}
+
 # lock PHP to minimum allowed version
 composer config platform.php 7.1.0
+cp composer.lock composer.lock.back || true
 composer update
 
 php box.phar compile -vv
 
-composer config --unset platform
+trap restorePlatform exit
```

Comment:
This trap should be before running `compile`, otherwise it has basically no effect here at the end.

Reply:
ha indeed :/